test/e2e/framework/service_util.go

```diff
@@ -1510,13 +1509,8 @@ func CheckAffinity(jig *ServiceTestJig, execPod *v1.Pod, targetIp string, target
 		if !shouldHold && !affinityHolds {
 			return true, nil
 		}
-		if shouldHold {
-			if !transitionState && !affinityHolds {
```

Comment:
_(without knowing much of the codebase)_ -- does the removal of this case lead to lower coverage?

Reply:
No. E2e test is behavior driven. And our expectation to the behavior doesn't match with the actual one. And this PR is the fix.